### PR TITLE
udevd: don't kill worker in manager_kill_workers when it's running

### PR DESCRIPTION
**Pull request by gaoyi1988**
If worker is running, kill worker may lead uevent unprocessed.

Fixes: https://github.com/systemd/systemd/issues/19039

### CODE REVIEW

File: src/udev/udevd.c [hunk 1]
```diff
@@ -1014,7 +1020,10 @@ static int on_worker(sd_event_source *s, int fd, uint32_t revents, void *userdat
                         continue;
                 }
 
-                if (worker->state != WORKER_KILLED)
+                if (worker->state == WORKER_KILLING) {
+                        worker->state = WORKER_KILLED;
+                        (void) kill(worker->pid, SIGTERM);
+                } if (worker->state != WORKER_KILLED)
```
Comment (yuwata) [inline src/udev/udevd.c]: `else if`

Comment (gaoyi1988) [reply src/udev/udevd.c]: @yuwata 
Modified. Thank you!

**Comment by yuwata** [comment]
`manager_kill_workers()` is also called in `manager_exit()`, in that case, I think we should all running workers must be killed soon. Otherwise, `systemctl stop systemd-udevd.service` may fail if running workers process events very slowly.

So, how about to introduce one extra boolean argument for `manager_kill_workers()` like that:
```
static void manager_kill_workers(Manager *manager, bool force) {
        struct worker *worker;

        assert(manager);

        HASHMAP_FOREACH(worker, manager->workers) {
                if (worker->state == WORKER_KILLED)
                        continue;

                if (worker->state == WORKER_RUNNING && !force) {
                        worker->state = WORKER_KILLING;
                        continue;
                }

                worker->state = WORKER_KILLED;
                (void) kill(worker->pid, SIGTERM);
        }
}
```
then call it with `force = true` on `manager_exit()`, and `false` at other places.

**Comment by yuwata** [comment]
(BTW, I really dislike the automatic reload of rules or builtins on change. But once I proposed to drop that feature, it was rejected as it may breaks backward compatibility...)

**Comment by gaoyi1988** [comment]
@yuwata 
updated

**Comment by yuwata** [comment]
LGTM. Thank you!

**Pull request merged**